### ISSUE TITLE
Add screen reader live region support

### ISSUE THREAD

**Issue by JasonStoltz** [elastic/search-ui#325]
This was proposed in the following PR: https://github.com/elastic/search-ui/pull/320.

After discussing, we believe screen reader live region support should be supported at the core level of search-ui, not just as part of the view.

To achieve that, we will do the following:

1. We will write notifications to a live region from SearchDriver, so that regardless of what view you are using, updates will always be written.
2. Notifications will be written as a side-effect of triggering actions. For example, after each new search request, we will write a message to the live region, something to the effect of 

    > "Showing 1 to 17 results out of 17, searching for "lake".

    This will be read aloud by the screen reader as updates occur.
3. Notification writing will be encapsulated in a helper, similar to `URLManager`. The helper will be doing explicit DOM manipulation in a DOM element that it will append to the body. This element is completely invisible and should be completely unobtrusive to users and developers.
4. The default behavior of the `SearchDriver` will be to write these live region updates directly to the DOM. However, they can be disabled by passing a boolean flag to the `SearchDriver` configuration. This also applies to the `SearchProvider` component, which simply passes configuration through to the `SearchDriver`.

    ```js
    SearchDriver({
      a11yNotifications: false
    })
    ```

    ```jsx
    <SearchProvider config={{ a11yNotifications: false }} />
    ```

5. In order to support internationalization concerns and customization, the notifications that are written to the live region should be configurable. They should be functions, rather than static strings, so that a developer can handle plurals ("result" vs "results").

    ```js
    SearchDriver({
      a11yNotificationMessages: {
        newSearchResults: ({searchTerm, begin, end, total}) => `Showing ${begin} to ${end} results out of ${total}, searching for "${searchTerm}"`.
      }
    })
    ```  

    Passing nothing for a message, omitting a message, or omitting the configuration entirely will use the default message for message.
    (Optional requirements) Passing `false` could disable that notification entirely

6. An action should be passed down that allows explicit updating of the live region. Something like `a11yNotification`.

    Example: If someone created a custom "RecommendResults" component, they could wire it up to call `a11yNotification`. Something like `a11y("Recommended results: here are some results...")`
